### PR TITLE
Changes to `probability` in drv.py and crv.py

### PR DESCRIPTION
**Pull request by akash9712**
#### Brief description of what is fixed or changed
1) drv.py - probability where conditions is of the form `Ne(RV, a)` is now
calculated as `1 - P(Eq(RV, a))`, similar to what is done in crv.py. This
results in significant reduction in time for calculating probabilities where
the RV is a poisson variable with a large mean. Eg. Y = Poisson('Y', 1000).

2) crv.py - condition to check if the domain resulting from the given
condition is a union of intervals, and calculate the required probability by
summing up the probabilites over each interval. Also removed a duplicate
import added in #14254.
#### Other comments
The check for `Union` was added in a commit in #14254, but was removed because I didn't think of any cases where it would be used. Test cases for the same have now been added.

### CODE REVIEW

**Review by normalhuman — APPROVED** [review]

File: sympy/stats/drv.py [hunk 1]
```diff
@@ -178,13 +178,17 @@ def restricted_domain(self, condition):
         return conditional_domain
 
     def probability(self, condition):
+        complement = isinstance(condition, Ne)
+        if complement:
+            condition = Eq(condition.args[0], condition.args[1])
```
Comment (normalhuman) [inline sympy/stats/drv.py]: Is there a test for this logic in drv.py?

Comment (akash9712) [reply sympy/stats/drv.py]: The tests at [L54](https://github.com/sympy/sympy/blob/master/sympy/stats/tests/test_discrete_rv.py#L54) and [L48](https://github.com/sympy/sympy/blob/master/sympy/stats/tests/test_discrete_rv.py#L48) in test_discrete_rv.py test for  condition of the type `Ne`. That should be a test for this logic.

File: sympy/stats/crv.py [hunk 2]
```diff
@@ -346,6 +346,10 @@ def probability(self, condition, **kwargs):
             # return S.Zero if `domain` is empty set
             if domain.set is S.EmptySet or isinstance(domain.set, FiniteSet):
                 return S.Zero if not cond_inv else S.One
+            if isinstance(domain.set, Union):
+                return sum(
+                     Integral(pdf(z), (z, subset), **kwargs) for subset in
+                     domain.set.args if isinstance(subset, Interval))
```
Comment (normalhuman) [inline sympy/stats/crv.py]: What role does `isinstance(subset, Interval)` have here? Anything that's not an Interval is simply ignored. Is this the right thing to do?

Comment (akash9712) [reply sympy/stats/crv.py]: Please correct me if I am wrong, but with the current implementation of `where`,  if `domain.set` is a union, each instance would be a `FiniteSet` or an `Interval`. Being continuous RVs, only the intervals contribute to the probability of the event. If `subset` is a `FiniteSet`, an error would be raised because of incorrect integration limits.
Should I do something like summing up the probability over each `subset`, and if an error is raised while caculating the integral, just add the unevaluated object to the total probability?

Comment (normalhuman) [reply sympy/stats/crv.py]: Thanks for explanation. Indeed, I can't think of a situation where `domain.set` would be a set of positive measure that's not a union of intervals. Things like a Cantor set of positive measure can hardly arise from `where`, and even if they did, they would not be representable by a Union.
